sirepo/auth/__init__.py

```diff
@@ -270,6 +270,13 @@ def is_premium_user(self):
             role=sirepo.auth_role.ROLE_PAYMENT_PLAN_PREMIUM,
         )
 
+    def local_part_user_name(self):
+        """If user_name is email address, return the local part.
+        Otherwise, just return user_name.
+        """
+
+        return self.logged_in_user_name().split("@")[0]
```

Comment:
This should assert `@` is in the name by checking there are exactly two parts. The name should reflect that it works for the logged_in_user, e.g. logged_in_user_email_local_part. It's a mouthful, but it describes what is going on. The coupling is a bit problematic. I think we might want instead to have a function logged_in_email and have it assert the email. Then a separate function that splits emails into parts eg in sirepo.util.

Reply:
That seems unnecessarily complicated for a function that is only called once. I also don't see how this suggested change would handle usernames that aren't email addresses. I just renamed the function to `logged_in_user_name_local_part`. [3a9f0d1](https://github.com/radiasoft/sirepo/pull/7027/commits/3a9f0d1399801354cea9e77a661893218636e2dd)